src/amo/components/RatingManager/index.js

```diff
@@ -81,23 +88,37 @@ export class RatingManagerBase extends React.Component {
       });
   }
 
-  renderLogInToRate() {
-    const { AuthenticateButton, addon, i18n, location } = this.props;
-    let authPrompt;
-    if (addon.type === ADDON_TYPE_EXTENSION) {
-      authPrompt = i18n.gettext('Log in to rate this extension');
-    } else if (addon.type === ADDON_TYPE_THEME) {
-      authPrompt = i18n.gettext('Log in to rate this theme');
-    } else {
-      throw new Error(`Unknown extension type: ${addon.type}`);
+  getLogInPrompt(
+    { addonType }, { validAddonTypes = defaultValidAddonTypes } = {}
+  ) {
+    const { i18n } = this.props;
+    switch (addonType) {
+      case ADDON_TYPE_DICT:
+        return i18n.gettext('Log in to rate this dictionary');
+      case ADDON_TYPE_LANG:
+        return i18n.gettext('Log in to rate this language pack');
+      case ADDON_TYPE_SEARCH:
+        return i18n.gettext('Log in to rate this search engine');
+      case ADDON_TYPE_THEME:
+        return i18n.gettext('Log in to rate this theme');
+      case ADDON_TYPE_EXTENSION:
+      default:
+        if (!validAddonTypes.includes(addonType)) {
+          throw new Error(`Unknown extension type: ${addonType}`);
+        }
+        return i18n.gettext('Log in to rate this extension');
```

Comment:
I understood this after a second but it might be more immediately obvious (and technically a bit more accurate) as:

```js
case ADDON_TYPE_EXTENSION:
  return i18n.gettext('Log in to rate this extension');
default:
  if (!validAddonTypes.includes(addonType)) {
    throw new Error(`Unknown extension type: ${addonType}`);
  }
  return i18n.gettext('Log in to rate this add-on');
```

And maybe even log that it's an unknown extension type not covered here. If we had a new "theme" type that was valid but not handled here it would be called, specifically, an "extension" when our grab-bag term "add-on" is more appropriate.